src/coreclr/jit/lowerxarch.cpp

```diff
@@ -6335,38 +6335,49 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 {
                     if ((intrinsicId >= NI_FMA_MultiplyAdd) && (intrinsicId <= NI_FMA_MultiplySubtractNegatedScalar))
                     {
-                        bool supportsRegOptional = false;
+                        bool     supportsOp1RegOptional = false;
+                        bool     supportsOp2RegOptional = false;
+                        bool     supportsOp3RegOptional = false;
+                        unsigned resultOpNum            = 0;
+                        LIR::Use use;
 
-                        if (IsContainableHWIntrinsicOp(node, op3, &supportsRegOptional))
+                        if (BlockRange().TryGetUse(node, &use))
                         {
-                            // 213 form: op1 = (op2 * op1) + [op3]
-                            MakeSrcContained(node, op3);
+                            resultOpNum = node->GetResultOpNumForFMA(use.User(), op1, op2, op3);
                         }
-                        else if (IsContainableHWIntrinsicOp(node, op2, &supportsRegOptional))
+
+                        // Prioritize Containable op. Check if any one of the op is containable first.
+                        // Set op regOptional only if none of them is containable.
+
+                        if (resultOpNum != 1 && IsContainableHWIntrinsicOp(node, op1, &supportsOp1RegOptional) &&
+                            !HWIntrinsicInfo::CopiesUpperBits(intrinsicId))
```

Comment:
Just writing out my own thoughts here, I don't think there is anything that strictly needs to be done for the PR. It could instead be handled in a follow up if we thought it was beneficial.

Consider the scenario where more than one operand is last use. Today, https://github.com/dotnet/runtime/pull/58196/files#diff-fba867eda1c745875748370c59da2195c3823b85a17d199444068d93f91c749bR21856-R21861 will just return the index of the first operand that's last use.

For this example, lets consider that all are last use and only `op1` is coming from memory. In this scenario, `GetResultOpNumForFMA` will return `1` and this check will skip the containment logic for `op1` because it was also last use, where-as in practice we also have `op2` and `op3` that are last use and so should be accounting for that and allowing `op1` to be contained.